Examples/Framework/include/ActsExamples/EventData/SimSpacePoint.hpp

```diff
@@ -8,28 +8,73 @@
 
 #pragma once
 
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Common.hpp"
+#include "ActsExamples/EventData/Index.hpp"
+
+#include <cmath>
+#include <vector>
+
 namespace ActsExamples {
-struct SimSpacePoint {
-  // Hit id
-  unsigned int m_index;
+
+/// Space point representation of a measurement suitable for track seeding.
+class SimSpacePoint {
+ public:
+  template <typename position_t, typename covariance_t>
+  SimSpacePoint(const Eigen::MatrixBase<position_t>& pos,
+                const Eigen::MatrixBase<covariance_t>& cov,
+                Index measurementIndex)
+      : m_x(pos[Acts::ePos0]),
+        m_y(pos[Acts::ePos1]),
+        m_z(pos[Acts::ePos2]),
+        m_r(std::hypot(m_x, m_y)),
+        //      r = sqrt(x² + y²)
```

Comment:
Yes, you are right. I added two changes:

1. The space point constructor takes the `rho` and `z` variance as input (instead of the full global covariance). The space point should not care how it was computed. Depending on the circumstance, it might be a fixed value or a simple lookup table and we should not presume any particular source here.
2. The `rho`/`z` variance is now computed by the space point maker taking into account all possible correlations.